docs/implementation.html

```diff
@@ -102,6 +102,56 @@ <h5><a id="recordheader" href="#recordheader">5.4.2.1 Record Header</a></h5>
     <p>We use the same varint encoding as Protobuf. More information on the latter can be found <a href="https://developers.google.com/protocol-buffers/docs/encoding#varints">here</a>. The count of headers in a record
     is also encoded as a varint.</p>
 
+    <h4><a id="messageset" href="#messageset">5.3.3 Old Message Format</a></h4>
+    <p>
+        Prior to Kafka 0.11 messages were transferred and stored in a format called MessageSets. In MessageSets each message has its own metadata. Note that although MessageSets are formally arrays,
+        they are not preceded by an int32 array size like other array elements in the protocol.
+    </p>
+
+    <b>Message Set:</b><br>
+    <p><pre class="brush: java;">
+    MessageSet (Version: 0) => [offset message_size message]
+        offset => INT64
+        message_size => INT32
+        message => crc magic_byte attributes key value
+            crc => INT32
+            magic_byte => INT8
+            attributes => INT8
+            key => BYTES
+            value => BYTES
+    </pre></p>
+    <p><pre class="brush: java;">
+    MessageSet (Version: 1) => [offset message_size message]
+        offset => INT64
+        message_size => INT32
+        message => crc magic_byte attributes key value
+            crc => INT32
+            magic_byte => INT8
+            attributes => INT8
+            timestamp =>INT64
+            key => BYTES
+            value => BYTES
+    </pre></p>
+    <p>
+        In versions prior to Kafka 0.10 the magic byte value was 0. Message format version 1 was introduced with timestamp support in version 0.10 .
+    <ul>
+        <li>Similarly to version 2 above, the lowest bits of attributes mean compression type.</li>
```

Comment:
Rather than describing the attributes in prose, could we represent them in the schema as we did above for the old message format:
```
 		attributes: int16
 			bit 0~2:
 				0: no compression
 				1: gzip
 				2: snappy
 				3: lz4
 			bit 3: timestampType
 			bit 4: isTransactional (0 means not transactional)
 			bit 5: isControlBatch (0 means not a control batch)
 			bit 6~15: unused
```

The explanation of the use of the timestamp type is useful though.